File: src/main/java/com/harshit/pharmacy/returns/service/ReturnService.java
package com.harshit.pharmacy.returns.service;

import com.harshit.pharmacy.returns.dto.CreateReturnRequest;
import com.harshit.pharmacy.returns.dto.ReturnResponse;
import com.harshit.pharmacy.returns.dto.UpdateReturnStatusRequest;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.transaction.annotation.Transactional;

public interface ReturnService {

    ReturnResponse createReturn(CreateReturnRequest request);

    Page<ReturnResponse> getMyReturns(Pageable pageable);

    @Transactional(readOnly = true)
    Page<ReturnResponse> getReturnsByStatus(String status, Pageable pageable);

    ReturnResponse updateReturnStatus(Integer returnId, UpdateReturnStatusRequest request);

}

File: pharmacy-frontend/src/services/returnService.js
import api from "../api/api";

export async function getReturnsByStatus({ status, page, size }) {
    const response = await api.get("/admin/returns", { params: { status, page, size } });
    return response.data;
}

export async function updateReturnStatus(returnId, returnStatus) {
    const response = await api.patch(`/admin/returns/${returnId}/status`, { returnStatus });
    return response.data;
}

export async function createReturn(request) {
    const response = await api.post("/returns", request);
    return response.data;
}

export async function getMyReturns(page = 0, size = 10) {
    const response = await api.get("/returns", {params: {page,  size}});
    return response.data;
}

